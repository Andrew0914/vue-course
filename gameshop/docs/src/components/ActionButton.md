# ActionButton

> A button to general purpouse

## Props

| Prop name | Description                  | Type   | Values | Default |
| --------- | ---------------------------- | ------ | ------ | ------- |
| event     | The event name to be emitted | string | -      | "click" |

## Events

| Event name  | Type | Description   |
| ----------- | ---- | ------------- |
| custom name |      | Custom event. |

## Slots

| Name    | Description                           | Bindings |
| ------- | ------------------------------------- | -------- |
| default | Use this slot to display label button |          |
| icon    | Use this slot to set icon button      |          |
