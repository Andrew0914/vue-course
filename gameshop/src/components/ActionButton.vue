<template>
  <button class="action" @click="emit">
    <!-- @slot Use this slot to display label button -->
    <slot />
    <!-- @slot Use this slot to set icon button -->
    <slot name="icon" />
  </button>
</template>

<script>
/**
 * A button to general purpouse
 * @displayName ActionButton
 */
export default {
  name: "ActionButton",
  props: {
    /**
     * The event name to be emitted
     */
    event: {
      type: String,
      default: "click"
    }
  },
  methods: {
    emit() {
      /**
       * Custom event.
       *
       * @event custom name
       */
      this.$emit(this.event);
    }
  }
};
</script>

<style scoped>
.action {
  padding: 8px;
  border-radius: 3px;
  background-color: rgb(251, 188, 4);
  display: flex;
  justify-content: space-between;
  align-items: center;
  min-width: 100px;
  margin: 8px 0;
  outline: none;
  border: none;
  cursor: pointer;
  font-size: 16px;
}

.action img {
  margin: auto 0;
  width: 16px;
  height: 16px;
  max-height: 16px;
  max-width: 16px;
}
</style>