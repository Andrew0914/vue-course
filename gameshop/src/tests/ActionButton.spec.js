import { mount } from "@vue/test-utils";
import ActionButton from "../components/ActionButton.vue";

it('Test button event', () => {
    const actionButton = mount(ActionButton, {propsData: {event: "test"}})
    const button = actionButton.find("button")
    button.trigger('click')

    expect(actionButton.emitted().test).toBeTruthy()
})

it("Insert content in slots", () => {
    const actionButton = mount(ActionButton, {propsData: {event: "test"}, slots: {
        icon: "<img/>",
        default: "HELLO"
    }})

    expect(actionButton).toMatchSnapshot()
})