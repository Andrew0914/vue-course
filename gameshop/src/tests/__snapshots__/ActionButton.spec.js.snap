// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Insert content in slots 1`] = `<button class="action">HELLO <img></button>`;
